corgidrp/astrom.py

```diff
@@ -0,0 +1,387 @@
+
+    Returns: 
+        platescale (float): the image platescale [mas/pixel]
+    
+    """
+    # load in the image data and source information
+    if type(image) != np.ndarray:
+        raise TypeError("Image type must be numpy.ndarray")
+
+    if type(source_guesses) != astropy.table.Table:
+        raise TypeError("source_guesses must be an astropy table with columns \'x\',\'y\'")
+    else:
+        guesses = source_guesses
+
+    if type(skycoords_) != astropy.table.Table:
+        raise TypeError("skycoords_ must be an astropy table with columns \'RA\',\'DEC\'")
+    else:
+        skycoords_table = skycoords_
+        skycoords = SkyCoord(ra = skycoords_table['RA'], dec= skycoords_table['DEC'], unit='deg', frame='icrs')
+    
+    # create 1_000 random combinations of stars
+    all_combinations = list(compute_combinations(guesses['x']))
+    rand_inds = np.random.randint(low=0, high=len(all_combinations), size=1000)
+    combo_list = np.array(all_combinations)[rand_inds]
+
+    # gather all the skycoord separations for all combinations
+    seps = np.empty(len(combo_list))
+    for c, i in zip(combo_list, range(len(combo_list))):
+        star1 = skycoords[c[0]]
+        star2 = skycoords[c[1]]
+
+        sep = star1.separation(star2).mas
+        seps[i] = sep
+
+    # find the separations in pixel space on the image between all combinations
+    pixseps = np.empty(len(combo_list))
+    for c, i in zip(combo_list, range(len(combo_list))):
+        star1 = guesses[c[0]]
+        star2 = guesses[c[1]]
+
+        xguess = star2['x'] - star1['x']
+        yguess = star2['y'] - star1['y']
+        
+        xoff, yoff = measure_offset(image, xstar_guess=star1['x'], ystar_guess=star2['x'], xoffset_guess= xguess, yoffset_guess= yguess)
+
+        pixsep = np.sqrt(np.power(xoff,2) + np.power(yoff,2))
+        pixseps[i] = pixsep
+
+    # estimate the platescale from each combination and find the mean
+    platescales = seps / pixseps
+    platescale = np.mean(platescales)
+
+    return platescale
+
+def angle_between(pos1, pos2):
+    """ 
+    Used to find the angle from North counterclockwise between two sources in an image.
+    
+    Args:
+        pos1 (tuple): the (x,y) position of the first target
+        pos2 (tuple): the (x,y) position of the second target
+    
+    Returns:
+        angle (float): the angle between image north and true north
+    """
+    
+    xdif = pos2[0] - pos1[0]
+    ydif = pos2[1] - pos1[1]  
+
+    if xdif<0:
+        if ydif<0:
+            angle = np.pi - np.arctan(xdif/ydif)
+        else:
+            angle = - np.arctan(xdif/ydif)
+    else:
+        if ydif<0:
+            angle = - np.arctan(xdif/ydif) + np.pi
+        else: 
+            angle = (2*np.pi) - np.arctan(xdif/ydif)
+            
+    return angle * 180/np.pi
+
+def compute_northangle(image, source_guesses, center_coord, skycoords_):
```

Comment:
I think the north angle and platescale computation can be combined together, since they both use the same structure (just look at a different measurable). This should simplify and speed up the code